sdk/src/test/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/connector/ApiConnectorTest.java

```diff
@@ -391,4 +410,39 @@ private JSONObject createPaymentBrandResponse() {
 
         return jsonObject;
     }
+
+    private JSONObject getOrderStatusResponse() {
+        JSONObject jsonObject = new JSONObject();
+        jsonObject.put("merchantOrderId", "25da863a-60a5-475d-ae47-c0e4bd1bec31");
+        jsonObject.put("id", "ORDER1");
+        jsonObject.put("poiId", "1");
+        jsonObject.put("orderStatus", "COMPLETED");
+        jsonObject.put("statusLastUpdatedAt", "2000-01-01T00:00:00.000-0200");
+        jsonObject.put("totalAmount", getJsonMoney(Currency.EUR, 100));
+
+        JSONObject firstTransaction = getTransactionObject("1", 100L, true);
+        JSONObject secondTransaction = getTransactionObject("2", 200L, true);
+        JSONObject thirdTransaction = getTransactionObject("3", 300L, false);
+        jsonObject.put("transactions", new JSONArray(Arrays.asList(firstTransaction)));
+        return jsonObject;
+    }
+
+    private JSONObject getJsonMoney(Currency currency, long amountInCents) {
+        JSONObject object = new JSONObject();
+        object.put("valueInMinorUnits", String.valueOf(amountInCents));
+        object.put("currency", currency);
+        return object;
+    }
+
+    private JSONObject getTransactionObject(String id, Long amount, boolean withConfirmedAmount) {
```

Comment:
withConfirmedAmount in never used so it can be removed from method. Also after you include 'REFUND' in TransactionType enum it would suggest to replace withConfirmedAmount with a transactionType and that  among this three transactions we have at least one with 'REFUND' type also.

Reply:
I would  also suggest to include transactionStatus in a method signature so we pass it also. This way we can have other transaction status beside 'SUCCESS' so it reflex a real scenario.

Reply:
Both done.